daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -191,7 +191,7 @@ private[speedy] sealed abstract class SBuiltin(val arity: Int) {
       i: Int,
   ): (TypeConName, SRecord) =
     args.get(i) match {
-      case SAnyContract(tyCon, value) => (tyCon, value)
+      case SAny(_, record: SRecord) => (record.id, record)
```

Comment:
```suggestion
      case SAny(TyCon(tyCon), record: SRecord) => (tyCon, record)
```
Please do not rely on the type in the record.

Reply:
ok, I will see what I can do here.

Reply:
done